chain/indexer/integrated/processor/state.go

```diff
@@ -394,6 +387,16 @@ func (sp *StateProcessor) startActor(ctx context.Context, current, executed *typ
 	return taskNames
 }
 
+func genErrResult(ctx context.Context, err error, name string, start time.Time) *Result {
+	stats.Record(ctx, metrics.ProcessingFailure.M(1))
+	return &Result{
+		Task:        name,
+		Error:       err,
+		StartedAt:   start,
+		CompletedAt: time.Now(),
+	}
+}
+
```

Comment:
Is there any particular reason why this was made into a function? It's easier to read and reason when it was in `sp.startActor`.